src/routes/v2/projects.rs

```diff
@@ -2558,3 +2561,214 @@ pub async fn delete_from_index(
+        href: format!("{self_addr}/project/{project_slug}/feed"),
+        rel: "self".to_string(),
+        mime_type: Some("application/atom+xml".to_string()),
+        title: Some(project.title.to_string()),
+        hreflang: None,
+        length: None,
+    });
+    feed.icon = project.icon_url;
+    feed.id = format!("{site_url}/mod/{project_id}");
+    feed.updated = project.updated.fixed_offset();
+    feed.rights = Some(project.license.name.into());
+    for category in &project.categories {
+        feed.categories.push(atom_syndication::Category {
+            term: category.to_string(),
+            label: Some(category.to_string()),
+            scheme: None,
+        });
+    }
+    for version in &versions {
+        let version_id = to_base62(version.id.0);
+        let version_slug = if versions
+            .iter()
+            .filter(|v| v.version_number == version.version_number)
+            .count()
+            == 1
+        {
+            &version.version_number
+        } else {
+            &version_id
+        };
+        let link = format!("{site_url}/mod/{project_slug}/version/{version_slug}");
+        let mut entry = atom_syndication::Entry::default();
+        entry.id = format!("{site_url}/mod/{project_id}/version/{version_id}");
+        entry.title = atom_syndication::Text::plain(&version.name);
+        entry.updated = version.date_published.fixed_offset();
+        entry.published = Some(version.date_published.fixed_offset());
+        let rendered_changelog = comrak::markdown_to_html(&version.changelog, &markdown_options);
+        if rendered_changelog.trim().is_empty() {
+            entry.summary = Some(atom_syndication::Text::plain(&version.name));
+            entry.content = Some(atom_syndication::Content {
+                src: Some(link.clone()),
+                content_type: Some("text/html".to_string()),
+                base: None,
+                lang: None,
+                value: None,
+            });
+        } else {
+            entry.content = Some(atom_syndication::Content {
+                base: Some(link.clone()),
+                content_type: Some("html".to_string()),
+                value: Some(rendered_changelog.to_string()),
+                lang: None,
+                src: None,
+            });
+        }
+        entry.links.push(atom_syndication::Link {
+            href: link.clone(),
+            rel: "alternate".to_string(),
+            mime_type: Some("text/html".to_string()),
+            title: Some(version.name.to_string()),
+            hreflang: None,
+            length: None,
+        });
+        entry.authors = match users.iter().find(|u| u.id == version.author_id) {
+            Some(u) => vec![atom_syndication::Person {
+                name: u.username.clone(),
+                uri: Some(format!("{site_url}/user/{}", &u.username)),
+                email: None,
+            }],
+            None => authors.clone(), // this should be unreachable
```

Comment:
The submitter could be somebody who was kicked from the team

Reply:
That's handled already! The `member_user_ids` array includes all team members and organisation members, as well as all submitters for all versions. See this code (lines 2612-2616) that loops through versions to fill up the array:
```rust
    for v in &versions {
        if !member_user_ids.contains(&v.inner.author_id) {
            member_user_ids.push(v.inner.author_id);
        }
    }
```

Reply:
Oh nice, I missed that :sweat_smile: